godel_path_planning/src/trajectory_planning.cpp

```diff
@@ -0,0 +1,188 @@
+            0.0, angle_discretization));
+  }
+
+  // Create an axial trajectory pt from a given tf transform
+  descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization)
+  {
+    Eigen::Affine3d eigen_pose;
+    tf::poseTFToEigen(nominal,eigen_pose);
+    Eigen::Vector3d rpy =  eigen_pose.rotation().eulerAngles(0,1,2);
+
+    double rx = rpy(0);
+    double ry = rpy(1);
+    double rz = rpy(2);
+    double x = eigen_pose.translation()(0);
+    double y = eigen_pose.translation()(1);
+    double z = eigen_pose.translation()(2);
+
+    return pointAxisTrajectoryFactory(x, y, z , rx, ry, rz, discretization);
+  }
+
+  // Translates a point relative to a reference pose to an absolute transformation
+  // Also flips the z axis of the orientation to point INTO the plane specified by the
+  // reference frame.
+  tf::Transform createNominalTransform(const geometry_msgs::Pose& ref_pose, const geometry_msgs::Point& point)
+  {
+    // To plane
+    tf::Transform marker_pose;
+    tf::poseMsgToTF(ref_pose, marker_pose);
+    // From plane to point
+    tf::Transform to_point = tf::Transform::getIdentity();
+    to_point.setOrigin(tf::Vector3(point.x, point.y, point.z));
+    // Reverse orientation of z axis
+    tf::Quaternion quat;
+    quat.setEuler(0.0, 3.14159, 0.0); // yaw, pitch roll
+    tf::Transform flip_z (quat);
+    // Calculate transform
+    tf::Transform in_world = marker_pose * to_point * flip_z;
+
+    return in_world;
+  }
+
+  bool populateTrajectoryMsg(const std::vector<descartes_core::TrajectoryPtPtr>& solution,
+                             const descartes_core::RobotModel& robot_model,
+                             double interpoint_delay,
+                             double time_offset,
+                             trajectory_msgs::JointTrajectory& trajectory)
+  {
+    typedef std::vector<descartes_core::TrajectoryPtPtr>::const_iterator JointSolutionIterator;
+    
+    // For calculating the time_from_start field of the trajectoryPoint
+    ros::Duration time_from_start(time_offset);
+
+    for (JointSolutionIterator it = solution.begin(); it != solution.end(); ++it)
+    {
+      // Retrieve actual target joint angles from the polymorphic interface function
+      std::vector<std::vector<double> > joint_angles;
+      it->get()->getJointPoses(robot_model, joint_angles);
+      const std::vector<double>& sol = joint_angles[0];
+      
+      trajectory_msgs::JointTrajectoryPoint point;
+      point.positions = sol;
+      point.velocities.resize(sol.size(), 0.0); // Fill extra fields with zeroes for now
+      point.accelerations.resize(sol.size(), 0.0);
+      point.effort.resize(sol.size(), 0.0);
+      point.time_from_start = time_from_start;
+
+      // add trajectory point to array
+      trajectory.points.push_back(point);
+
+      // increment time so far by next duration
+      time_from_start += ros::Duration(interpoint_delay);
+    }
+
+    return true;
```

Comment:
You could return false here if a joint solution array can't be retrieved from the trajectory point.  Although this will likely get caught much earlier in the planning process.